pkg/apis/extensions/validation/validation.go

```diff
@@ -886,3 +887,14 @@ func ValidatePodSecurityPolicyUpdate(old *extensions.PodSecurityPolicy, new *ext
 	allErrs = append(allErrs, ValidatePodSecurityPolicySpec(&new.Spec, field.NewPath("spec"))...)
 	return allErrs
 }
+
+func ValidateSelectorImmutability(update *metav1.LabelSelector, old *metav1.LabelSelector) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if update == nil || old == nil {
+		return allErrs
+	}
+	if !reflect.DeepEqual(update, old) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("selector"), update, "selector must not be changed after update"))
```

Comment:
assuming the field path here isn't good, pass in the field path?

Reply:
Sure! let's make `spec.selector` a parameter.